scripts/runtests.sh

```diff
@@ -231,6 +231,29 @@ grep -q 'finish: subiquity/Install/install/postinstall/run_unattended_upgrades:
 
 # The OOBE doesn't exist in WSL < 20.04
 if [ "${RELEASE%.*}" -ge 20 ]; then
+    # Test TCP connectivity (system_setup only)
+    clean
+    port=50321
+    LANG=C.UTF-8 timeout --foreground 60 \
+        python3 -m system_setup.cmd.server --dry-run --tcp-port=$port &
+    subiquity_pid=$!
+    next_time=3
+    until [ $next_time -eq 0 ] || [ ! -z "$(ss -Hlt sport = $port)" ]; do
+        sleep $(( next_time-- ))
+    done
+    if [ $next_time -eq 0 ]; then
+        echo "Timeout reached before Subiquity TCP socket started listening"
+        exit 1
+    fi
+    curl "http://localhost:$port/meta/status"
```

Comment:
I won't require it for this PR, but a nice-to-have to consider - an integration test that connects on a non-loopback interface should never work.  As mentioned previously, I wish to be extremely certain that we don't listen on a public-facing interface.

(Once, when doing a windows server install, it was behaving weirdly, and I made the determination that the server had been compromised before I had even completed the install.  The internet can be scary.)

Reply:
Hey, @dbungert , do you think it would be worthy to add a middleware in system_setup to check each if  `request.remote == "127.0.0.1" `  and raise `HttpForbidden` if not?  Or is it too silly?

Reply:
I do not think that would be too silly.

Reply:
I'm asking because I don't know those attacks could be, but if they were like a reverse proxy, checking for request.remote would be misleading, since the requests reach the server coming from the proxy. In some sense the virtual adapters connecting Windows and Linux on WSL remind me of those reverse proxies.

Reply:
I loved this idea. Thank you!